tests/sentry/sentry_metrics/test_multiprocess_steps.py

```diff
@@ -288,24 +289,39 @@ def test_process_messages(mock_indexer) -> None:
             "project_id": 3,
         },
         "invalid_metric_name",
+        True,
+    ),
+    (
+        b"invalid_json_payload",
+        "invalid_json",
+        False,
     ),
 ]
 
 
-@pytest.mark.parametrize("invalid_payload, error_text", invalid_payloads)
-def test_process_messages_invalid_messages(invalid_payload, error_text, caplog) -> None:
+@pytest.mark.parametrize("invalid_payload, error_text, format_payload", invalid_payloads)
```

Comment:
this test case is pretty large and requires more knowledge of the system than we want in order to test the intended functionality. Is there a way to break either the test or the code down to make what's happening more obvious?

Reply:
I mean to do this earlier and then forgot but maybe filling in the doc string here would be the most helpful?